docs/API.md

```diff
@@ -104,133 +177,185 @@ gulp.task('somename', function() {
+### gulp.parallel(tasks, ...)
 
-2. In task "two" you add a hint telling the engine that it depends on completion of the first task.
+Takes a variable amount of task names or function and returns
+a function of the composed tasks or functions. When using task names,
+the task should already registered.
 
-So this example would look like this:
+When the returned function is executed, the tasks or functions will be executed in parallel,
+all being executed at the same time. If an error occurs, all execution will complete.
 
 ```js
-var gulp = require('gulp');
-
-// takes in a callback so the engine knows when it'll be done
-gulp.task('one', function(cb) {
-    // do stuff -- async or otherwise
-    cb(err); // if err is not null and not undefined, the run will stop, and note that it failed
+gulp.task('one', function(done) {
+  // do stuff
+  done();
 });
 
-// identifies a dependent task must be complete before this one begins
-gulp.task('two', ['one'], function() {
-    // task 'one' is done now
+gulp.task('two', function(done) {
+  // do stuff
+  done();
 });
 
-gulp.task('default', ['one', 'two']);
+gulp.task('default', gulp.parallel('one', 'two', function(done) {
+  // do more stuff
+  done();
+}));
 ```
 
+#### tasks
+Type: `Array`, `String` or `Function`
 
-### gulp.watch(glob [, opts], tasks) or gulp.watch(glob [, opts, cb])
+A task name, a function or an array of either.
 
-Watch files and do something when a file changes. This always returns an EventEmitter that emits `change` events.
 
-### gulp.watch(glob[, opts], tasks)
+### gulp.series(tasks, ...)
 
-#### glob
-Type: `String` or `Array`
+Takes a variable amount of task names or function and returns
+a function of the composed tasks or functions. When using task names,
+the tasks should already registered.
 
-A single glob or array of globs that indicate which files to watch for changes.
+When the returned function is executed, the tasks or functions will be executed
+in series, each waiting for the prior to finish. If an error occurs,
+execution will stop.
 
-#### opts
-Type: `Object`
+```js
+gulp.task('one', function(done) {
+  // do stuff
+  done();
+});
 
-Options, that are passed to [`gaze`](https://github.com/shama/gaze).
+gulp.task('two', function(done) {
+  // do stuff
+  done();
+});
+
+gulp.task('default', gulp.series('one', 'two', function(done) {
+  // do more stuff
+  done();
+}));
+```
 
 #### tasks
-Type: `Array`
+Type: `Array`, `String` or `Function`
 
-Names of task(s) to run when a file changes, added with `gulp.task()`
+A task name, a function or an array of either.
+
+
+### gulp.watch(glob[, opts], tasks)
+
+Watch files and do something when a file changes.
 
 ```js
-var watcher = gulp.watch('js/**/*.js', ['uglify','reload']);
-watcher.on('change', function(event) {
-  console.log('File ' + event.path + ' was ' + event.type + ', running tasks...');
-});
+gulp.watch('js/**/*.js', ['uglify', 'reload', function(done) {
+  // do something
+  done();
+}]);
```

Comment:
This isn't my decision to make, but I feel it probably shouldn't be encouraged to pass functions directly into `watch`. Sure, mention it below, but don't do it in a code example.

@phated @contra: thoughts?